### ISSUE TITLE
tensorflow static library of windows: missing few files

### ISSUE THREAD

**Issue by ignvinay** [tensorflow/tensorflow#59762]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1.  It must be a bug, a feature request, or a significant problem with the
    documentation (for small docs fixes please send a PR instead).
2.  The form below must be filled out.
3.  It shouldn't be a TensorBoard issue. Those go
    [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: Its tensorflow provided code as given in https://www.tensorflow.org/install/lang_c
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: windows 10
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**: Laptop
-   **TensorFlow installed from (source or binary)**: static library from https://www.tensorflow.org/install/lang_c
-   **TensorFlow version (use command below)**: 2.11
-   **Python version**: NA
-   **Bazel version (if compiling from source)**: NA
-   **GCC/Compiler version (if compiling from source)**: 12.2
-   **CUDA/cuDNN version**:NA
-   **GPU model and memory**: NA
-   **Exact command to reproduce**: Took static library from tensorflow ebsite from link below : https://www.tensorflow.org/install/lang_c
Have taken windows version of https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-windows-x86_64-2.11.0.zip (CPU version)
unzipped and created hello_tf.c as mentioned in same webpage.
Compiled the hello_tf.c as below : 
gcc hello_tf.c -I D:\tensorflow_lib\include -lD:\tensorflow_lib\lib\tensorflow -o hello_tf

Got error as 
_In file included from hello_tf.c:2:
D:\tensorflow_lib\include/tensorflow/c/c_api.h:23:10: fatal error: tensorflow/c/tf_buffer.h: No such file or directory
   23 | #include "tensorflow/c/tf_buffer.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated._

Then took tf_buffer.h from github repo , to see if the error is only dependent on tf_buffer.h, but then got another error : 

_D:\tensorflow_lib>gcc hello_tf.c -I D:\tensorflow_lib\include -lD:\tensorflow_lib\lib\tensorflow -o hello_tf
In file included from D:\tensorflow_lib\include/tensorflow/c/tf_tstring.h:19,
                 from D:\tensorflow_lib\include/tensorflow/c/c_api.h:27,
                 from hello_tf.c:2:
D:\tensorflow_lib\include/tensorflow/core/platform/ctstring.h:19:10: fatal error: tensorflow/tsl/platform/ctstring.h: No such file or directory
   19 | #include "tensorflow/tsl/platform/ctstring.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated._

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

Took static library from tensorflow ebsite from link below : https://www.tensorflow.org/install/lang_c
Have taken windows version of https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-windows-x86_64-2.11.0.zip (CPU version)
unzipped and created hello_tf.c as mentioned in same webpage.
Compiled the hello_tf.c as below : 
gcc hello_tf.c -I D:\tensorflow_lib\include -lD:\tensorflow_lib\lib\tensorflow -o hello_tf

Got error as 
_In file included from hello_tf.c:2:
D:\tensorflow_lib\include/tensorflow/c/c_api.h:23:10: fatal error: tensorflow/c/tf_buffer.h: No such file or directory
   23 | #include "tensorflow/c/tf_buffer.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated._

Then took tf_buffer.h from github repo , to see if the error is only dependent on tf_buffer.h, but then got another error : 

_D:\tensorflow_lib>gcc hello_tf.c -I D:\tensorflow_lib\include -lD:\tensorflow_lib\lib\tensorflow -o hello_tf
In file included from D:\tensorflow_lib\include/tensorflow/c/tf_tstring.h:19,
                 from D:\tensorflow_lib\include/tensorflow/c/c_api.h:27,
                 from hello_tf.c:2:
D:\tensorflow_lib\include/tensorflow/core/platform/ctstring.h:19:10: fatal error: tensorflow/tsl/platform/ctstring.h: No such file or directory
   19 | #include "tensorflow/tsl/platform/ctstring.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated._

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by gaikwadrahul8** [comment]
Hi, @ignvinay 

Apologize for the delay and I was able to replicate same issue on Google Colan which uses Ubuntu OS so it's working as expected, here is [gist-file ](https://colab.research.google.com/gist/gaikwadrahul8/284cad934c3d2456efbe6184af22f682/lang_c.ipynb)so we'll try on Windows 10 and we'll let you know soon and Thank you for noticing this issue. Thank you!

**Comment by zarzaro** [comment]
Hi, i got the same issue on Win10 and Visual Studio 2019 and the prebuild Tensorflow lib in Version 2.11 (https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-windows-x86_64-2.11.0.zip). 

Also tried older versions of TF like 2.6 and there is the same problem. 

There is no tensorflow/c/tf_buffer.h. The file doesnt exist.

I downloaded the linux version and here the tf_buffer.h exists.


Please help :(

**Comment by ignvinay** [comment]
Hi Zarzaro and gaikwadrahul,
       I could able to get it working. The include folder is missing two folders which are required for building.
These two folders i picked it from linux version of package as include files are more of header files, which would be independent of OS.

Here is comparision: (right side of the figure is my working directory).
![image](https://user-images.githubusercontent.com/47418607/223139078-f666685c-8977-4968-99cf-3c0d38bffe2d.png)

JUst incase if you still need whole include folder, i am attaching here for your reference. Please replace this include folder with include folder of tensorflow.
[include.zip](https://github.com/tensorflow/tensorflow/files/10899294/include.zip)

**Comment by zarzaro** [comment]
Hm okay at least that seems to be a solution. But its not straight forward. Is that a build bug? Are we the first two persons that are using the prebuilt lib from tensorflow for Windows? A prebuilt lib should be straight forward to use in my opinion. 

I even tried to build the dll etc from tensorflow since since 3 days without final success. The bazel build of dll, lib and headers is finally completed successfully but i get errors in Visual Studio as well when i try to run a simple hello world project. 

It seems Tensorflow is not meant to be used in c++. Very little documentation for this inconvenient build process.

**Comment by MarkDaoust** [comment]
Yeah, it's cool that there's a workaround, but it would be much better if we could find the script that makes these, and fix it.

**Comment by esohns** [comment]
also missing from the include "patch" is file "tensorflow/tsl/c/tsl_status.h" (there might be others)
